### ISSUE TITLE
Deleting a video does not delete the thumbnail and subtitle files

### ISSUE THREAD

**Issue by GlassedSilver** [Tzahi12345/YoutubeDL-Material#107]
And when I try to delete them myself with Windows Explorer I am getting a permissions error.

Fixing both the permissions as well as the deletion by youtubedl-material itself would be extremely helpful.

**Comment by Tzahi12345** [comment]
Sounds like a simple fix, just check if the thumbnails/subs exist and delete if so. Thanks for the note!

I'll also look into the permissions -- probably just as simple as above: check if they exist, and if so, update permissions.

**Comment by Tzahi12345** [comment]
I have good news and bad news. Good news is, I implemented permission updates and auto deletion for thumbnails in the [latest commit](https://github.com/Tzahi12345/YoutubeDL-Material/commit/55378521347bb3e7f897d9a75fb5b9e70dfbbb04).

Bad news is, I couldn't do the same for subtitles. We just don't keep track of that information (that we downloaded it), and as such we don't know which files to delete.

I could have done something fancy like using wildcards to delete all files that match `<name_of_video>.*.vtt` but there are some nasty edge cases there were subtitles you wanted to keep get deleted because the wild card coincidentally matches other subtitles for different videos.

So until we get a system described [here](https://github.com/Tzahi12345/YoutubeDL-Material/issues/139#issuecomment-650616184) this won't be super easy. If you think of any workarounds, let me know!

**Comment by GlassedSilver** [comment]
I agree that this should just get lumped into the approach you linked. It's the cleanest way to keep track of files and we need awareness of what we store anyhow eventually. Looking for and implementing a workaround for an issue that isn't even mid-priority would be overblown given that the item awareness isn't too far away anyhow?

I most certainly appreciate your drive to resolve this somehow using a workaround in the meantime, but keeping this issue here low priority for now is perfectly fine! :)
